docs/developer-docs/smart-contracts/compile.mdx

```diff
@@ -0,0 +1,73 @@
+---
+keywords: [beginner, tutorial, build]
+---
+
+import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
+
+# Compile
+
+<MarkdownChipRow labels={["Beginner", "Tutorial"]} />
+
+## Overview
+
+After you have written the code for your project's canisters, you need to compile the code into a WebAssembly module before it can be deployed on ICP. To build your canister's Wasm module, you can use the `dfx build` command. The `dfx build` command looks for source code to compile for each canister configured under the `canisters` section in the `dfx.json` configuration file. It can be used to build a specific canister or all canisters defined in the project's `dfx.json` file.
```

Comment:
I suggest that we link to a good dfx.json example to help developers how to build a dfx.json and also link to the dfx.json schema doc as well: https://internetcomputer.org/docs/current/developer-docs/developer-tools/cli-tools/dfx-json-reference